### PR TITLE
Change NamedType definition syntax

### PR DESCRIPTION
**Pull request by adithyaselv**
Changes NamedType definitions to use `:=`

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Lexer/TokKind.cs [hunk 1]
```diff
@@ -303,6 +303,12 @@ public enum TokKind
         /// Start of body for user defined functions.
         /// <code>=></code>
         /// </summary>
-        DoubleBarrelArrow,
+        DoubleBarrelArrow,
+
+        /// <summary>
+        /// Colon immediately followed by equal.
+        /// <code>=></code>
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Lexer/TokKind.cs:310]: >>=> [](http://example.com/codeflow?start=17&length=3)

`:=`

File: src/strings/PowerFxResources.en-US.resx [hunk 2]
```diff
@@ -4115,6 +4115,10 @@
     <value>Named formula must be an expression.</value>
     <comment>This error message shows up when Named formula is not an expression. For example, a = ;</comment>
   </data>
+  <data name="ErrNamedType_MissingTypeLiteral" xml:space="preserve">
+    <value>Named type must be a TypeLiteral.</value>
```
Comment (CarlosFigueiraMSFT) [inline src/strings/PowerFxResources.en-US.resx:4119]: >TypeLiteral [](http://example.com/codeflow?start=32&length=11)

AFAIK we don't have a concept of "TypeLiteral" that is exposed to makers, so we should use the common form `type literal`

File: src/strings/PowerFxResources.en-US.resx [hunk 3]
```diff
@@ -4115,6 +4115,10 @@
...
+    <comment>This error message shows up when Named type is not a TypeLiteral. For example, T := ; T2 := 5;</comment>
```
Comment (CarlosFigueiraMSFT) [inline src/strings/PowerFxResources.en-US.resx:4120]: >TypeLiteral [](http://example.com/codeflow?start=66&length=11)

Can you also explain what a type literal is (i.e., give _valid_ definitions of a type)? This way the translators will have a better understanding of this term.

File: src/tests/Microsoft.PowerFx.Core.Tests.Shared/UserDefinedTypeTests.cs [hunk 4]
```diff
@@ -118,11 +118,11 @@ public void TestValidUDTCounts(string typeDefinition, int expectedDefinedTypesCo
...
-        [InlineData("X= Type({ f:Number, f:Number});", 1, "ErrNamedType_InvalidTypeDefinition")]
-        [InlineData("B = Type({ x: A }); A = Type(B);", 2, "ErrNamedType_InvalidCycles")]
-        [InlineData("B = Type(B);", 1, "ErrNamedType_InvalidCycles")]
-        [InlineData("Currency = Type({x: Text}); Record = Type([DateTime]); D = Type(None);", 2, "ErrNamedType_InvalidTypeName")]
+        [InlineData("Point := Type({ x: Number, y: Number }); Point := Type(Number);", 1, "ErrNamedType_TypeAlreadyDefined")]
+        [InlineData("X:= Type({ f:Number, f:Number});", 1, "ErrNamedType_InvalidTypeDefinition")]
+        [InlineData("B := Type({ x: A }); A := Type(B);", 2, "ErrNamedType_InvalidCycles")]
+        [InlineData("B := Type(B);", 1, "ErrNamedType_InvalidCycles")]
+        [InlineData("Currency := Type({x: Text}); Record := Type([DateTime]); D := Type(None);", 2, "ErrNamedType_InvalidTypeName")]
         public void TestUDTErrors(string typeDefinition, int expectedErrorCount, string expectedMessageKey)
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests.Shared/UserDefinedTypeTests.cs:126]: >TestUDTErrors [](http://example.com/codeflow?start=20&length=13)

Can you add a case where the expected message key is the new one (ErrNamedType_MissingTypeLiteral)?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Pull request merged**